map.js

```diff
@@ -11,70 +17,86 @@ let mapY = 0;
 }
- 
-//attaches width to width of iframe
-function iFrameCheck(){
-    if(document.getElementById("iframeContainer")){
-        document.getElementById("iframeContainer").onclick = resizeMap;
-        window.removeEventListener("mousemove", iFrameCheck); //trim listener, it's served it's purpose.
-        resizeMap();
-    }
-}
-
-//matches width of map to Iframe width
-function resizeMap(){
-    if(document.getElementById("iframeContainer")){
-        var ifc = document.getElementById("iframeContainer");
-
-        if(debug){
-            ifc.style.width = "1000px";
-            ifc.style.height = "25px";
-            wrapper.style.height = "580px";
-        }
-
-        wrapper.style.width = ifc.clientWidth + "px";
-        wrapper.style.top = (ifc.clientHeight + 48).toString() + "px";
-    }
-    drawMap();    
-}
 
 function drawMap(){
     //maybe useful for placing icons.
     var aspectw = 443;
     var aspecth = 362;
 
-    mapCanvasContext.drawImage(img, mapX, mapY, (img.width * zoomLevel), (img.height * zoomLevel), 
-                                    0, 0, img.width, img.height);
+    mapCanvasContext.drawImage(img_Map, mapX, mapY, (img_Map.width * zoomLevel), (img_Map.height * zoomLevel), 
+                                    0, 0, img_Map.width, img_Map.height);
+    //draw all map icons //TODO: add in overlay options
+    for(let i = 0; i < locArr.length;i++){
+        if(locArr[i].icon == "Ayleid") drawIcon(icon_Ayleid, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Camp") drawIcon(icon_Camp, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Cave") drawIcon(icon_Cave, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Fort") drawIcon(icon_Fort, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Gate") drawIcon(icon_Gate, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Inn") drawIcon(icon_Inn, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Landmark") drawIcon(icon_Landmark, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Mine") drawIcon(icon_Mine, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Settlement") drawIcon(icon_Settlement, locArr[i].approxX, locArr[i].approxY);
+        else if(locArr[i].icon == "Shrine") drawIcon(icon_Shrine, locArr[i].approxX, locArr[i].approxY);
+        else console.warn("Element at " + i +" has no icon.");
+    }
+}
+
+//give x/y as reg in game cords.
+function drawIcon(icon, iconX = 0.5, iconY = 0.5){
+    var MapW = img_Map.width;
+    var MapH = img_Map.height;
+    var iconWH = 36 / zoomLevel;
+    var worldW = 480000;
+    var worldH = 400000;
+
+    //convert worldspace loc to % loc
+    iconX = (iconX + worldW / 2) / worldW;
+    iconY = (-iconY + worldH / 2) / worldH;
+
+    //apply % loc to map x/y
+    mapCanvasContext.drawImage(icon, ((MapW * iconX) - mapX) / zoomLevel - iconWH, 
```

Comment:
instead of premultiplying iconWH, you can just divide the whole statement by zoomLevel here

Reply:
i refactored the whole world space to canvas space calculation into its own thing, and it now returns an object with wh (iconWH),  x, and y.

wh has been useful to use in placement of text/labels as well, I think it's still worth calculating it.

maybe we could move the calculation in the mapzoom function instead to save a bit of processing power, since that's when that changes anyways.